doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+
+.. _endhost_policy_selection:
+
+Exposing policies to the endhosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+This results in a set of *PolicyIdentifiers* per hop, which can then be used by the application, such as by defining an application parameter
+that then selects the policies to use on the path and hands these to the snet implementation, e.g. ``--fabridpolicy``. 
+
+
+``fabridpolicy`` parameter
+''''''''''''''''''''''''''''
+A custom language is used to make a selection out of the available paths and policies. The basic components of the language are as follows:
+
+* **Identifiers**
+
+  An identifier matches with a specific hop in the path and applies a policy to that hop. Parts of this hop identifier may be a wildcard,
+  such that the identifier can match with multiple hops in the path.
+  An identifier is structured as follows: 
+  ``ISD-AS#IGIF,EGIF@POLICY``,
+  where 
+
+  * ISD can be either the ISD number (e.g. ``1``), or a wildcard (``0``).
+  * AS can be either the AS number seperated by underscores (e.g. ``ff00_0_110``) or a wildcard (``0``).
+  * IGIF can be either the ingress interface number (e.g. ``42``), or a wildcard (``0``).
+  * EGIF can be either the egress interface number (e.g. ``41``), or a wildcard (``0``).
+  * POLICY can be either the policy to apply, where a local policy is denoted as ``L`` + the policy identifier (e.g. ``L100``) and a global policy
+    is denoted by ``G`` + the policy identifier (e.g. ``G100``), a wildcard (``0``), or a rejection ``REJECT``.
+    Rejection means that this path should not be chosen. 
+
+  When used in a query, the identifier evaluates to true when at least a single hop in the path matches the identifier.
+
+* **Concatenations**
+
+  Multiple identifiers can be combined by using a concatenation (and/or parentheses). Concatenations are created by the ``+`` symbol. 
+
+  Example:
+
+  ``(1-0#0,0@G300 + 1-0#0,0@G200)`` applies both policy G300 and policy G200.
+
+  When used in a query, a concatenation evaluates to true when *all* identifiers in the concatenation also evaluate to true.
+
+
+* **Queries**
+
+  You can query for the existence of a specific hop and/or policy through a query.
+  Queries are structured as follows: ``{ QUERY_EXPRESSION ? EXPRESSION_IF_TRUE : EXPRESSION_IF_FALSE}``.
+  The query expression is evaluated, and if an identifier matches with a specific hop, the ``expression_if_true`` branch is applied.
+  If no matches can be found in the path, the ``expression_if_false`` branch is applied.
+  Identifiers in the query expression are not applied, e.g. if a query expression queries for a specific policy, the specific policy
+  is not applied to the hops it matches, unless the same expression is also given under the ``expression_if_true`` branch.
+
+  Example:
```

Comment:
Highlight example somehow, e.g., as code block

Reply:
Also, no need to mention "With the path" (as this is not a full sentence).
Maybe add a full sentence before the example, instead of putting "With the path" on the same line as the example?

Reply:
Also, how is a "path" defined?
It seems it used slightly different syntax than what is described above?

Reply:
I added a sentence but I also think the path looks a bit strange, e.g. why do we have a "->" between the second and last AS but not the first and second, and the first AS in the path uses () for I assume no policy, but the description of an Identifier above expectes (0) for wildcard? @jeltevanbommel

Reply:
I added the missing -> 
The wildcard is for selecting policies, in this case I want to denote that there are no policies on that hop, for which a wildcard does not apply. I've added an explaining sentence.